LibreNMS/OS/Vrp.php

```diff
@@ -95,7 +97,38 @@ public function discoverEntityPhysical(): Collection
 
         return $inventory;
     }
+     public function discoverTransceivers(): Collection
+   {
+    // Get a map of ifIndex to port_id for proper association with ports
+    $ifIndexToPortId = $this->getDevice()->ports()->pluck('port_id', 'ifIndex');
+
+    // Walk through the MIB table for transceiver information
+    return SnmpQuery::cache()->walk('HUAWEI-ENTITY-EXTENT-MIB::hwOpticalModuleInfoTable')->mapTable(function ($data, $ifIndex) use ($ifIndexToPortId) {
+        // Skip inactive transceivers
+        if ($data['HUAWEI-ENTITY-EXTENT-MIB::hwEntityOpticalType'] === 'inactive') {
+            return null;
+        }
 
+        // Handle cases where required data might not be available (fallback to null)
+        $cable = $data['HUAWEI-ENTITY-EXTENT-MIB::hwEntityOpticalCableType'] ?? null;
+        $distance = $data['HUAWEI-ENTITY-EXTENT-MIB::hwEntityOpticalDistance'] ?? null;
+        $wavelength = $data['HUAWEI-ENTITY-EXTENT-MIB::hwEntityOpticalWaveLengthExact'] ?? null;
+
+        // Create a new Transceiver object with the retrieved data
+        return new Transceiver([
```

Comment:
You are missing the import for Transceiver :)

Reply:
`use App\Models\Transceiver;`

Reply:
Done